src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.CacheEntryState.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.Caching.Memory
+{
+    internal partial class CacheEntry
+    {
+        // this type exists just to reduce CacheEntry size by replacing many enum & boolean fields with one of a size of Int32
+        [StructLayout(LayoutKind.Explicit)]
+        private struct CacheEntryState
+        {
+            [FieldOffset(0)]
+            private byte _flags;
+            [FieldOffset(1)]
+            private byte _evictionReason;
+            [FieldOffset(2)]
+            private byte _priority;
+            [FieldOffset(3)]
+            private byte _reserved; // for future use
+
+            internal CacheEntryState(CacheItemPriority priority) : this() => _priority = (byte)priority;
```

Comment:
`: this()` doesn't seem necessary.

```suggestion
            internal CacheEntryState(CacheItemPriority priority) => _priority = (byte)priority;
```

Reply:
![obraz](https://user-images.githubusercontent.com/6011991/103653960-8ebd6480-4f65-11eb-8152-04248a6625ca.png)